### PR TITLE
AKMC-113: Add SADB Interfaces for MySQL/Inmemory configurability and basic implementation.

### PR DESCRIPTION
**Pull request by IbraheemYSaleh**

### CODE REVIEW

File: fsw/crypto_sadb/sadb_mariadb_admin_scripts/create_sadb.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
+CREATE DATABASE IF NOT EXISTS sadb;
```
Comment (szemerick) [inline fsw/crypto_sadb/sadb_mariadb_admin_scripts/create_sadb.sql:1]: Like the directory structure for the SADB / maria / SQL scripts.  We were just talking this week about doing something like this to keep things separated for the different SA implementations.

File: fsw/public_inc/crypto_error.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/* Copyright (C) 2009 - 2017 National Aeronautics and Space Administration. All Foreign Rights are Reserved to the U.S. Government.
```
Comment (szemerick) [inline fsw/public_inc/crypto_error.h:1]: Note for us - we need to update copyrights and change the email address to the new mailing list.

File: fsw/public_inc/sadb_routine.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * foreign persons.
+ */
+
+#ifndef CRYPTOLIB_SADB_ROUTINE_H
+#define CRYPTOLIB_SADB_ROUTINE_H
+
+#ifdef NOS3 //NOS3/cFS build is ready
+#include "common_types.h"
+#include "osapi.h"
+#else //Assume build outside of NOS3/cFS infrastructure
```
Comment (szemerick) [inline fsw/public_inc/sadb_routine.h:21]: Good assumption that the build is outside nos3/cFS by default.  We're still having discussions about how to ensure that we maintain the spacecraft build/functionality.  Right now, we're leaning toward CryptoLib being a generic lib, and we will go back and adapt for spacecraft.  (This is a role reversal for how CryptoLib was first designed.)

File: fsw/public_inc/sadb_routine.h [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+#include "common_types_minimum.h"
+#include "osapi_minimum.h"
+#endif
+
+#include "crypto_structs.h"
+
+typedef struct {
```
Comment (szemerick) [inline fsw/public_inc/sadb_routine.h:28]: Like this.  Clean and straightforward.

**Review by szemerick — APPROVED** [review]

**Comment by dccutrig** [comment]
Overall looked pretty good to me. Was able to pare down the diff that shows up on github with _git diff --ignore-all-space akmc-113..collab_main -- fsw/src/crypto.c_ Excited for this merge and more progress

File: fsw/src/sadb_routine_inmemory.template.c [hunk 5]
```diff
@@ -239,6 +251,39 @@ static int32 sadb_get_sa_from_spi(uint16 spi,SecurityAssociation_t** security_as
     return status;
 }
 
+static int32 sadb_get_operational_sa_from_gvcid(uint8 tfvn,uint16 scid,uint16 vcid,uint8 mapid,SecurityAssociation_t** security_association)
```
Comment (dccutrig) [inline fsw/src/sadb_routine_inmemory.template.c:258]: Nice, I like this and how it clears up the old SA stuff

**Comment by dccutrig** [comment]
Looks good - I like the direction the SA DB API is taking.

**Pull request merged**